src/lib/BUTTON/button.h

```diff
@@ -1,37 +1,89 @@
 #pragma once
 
-#include "targets.h"
+#include <functional>
 
-class button
+template <uint8_t PIN, bool IDLELOW>
+class Button
 {
 private:
-    static int buttonPin;
-    static bool activeHigh;
+    // Constants
+    static constexpr uint32_t MS_DEBOUNCE = 25;       // how long the switch must change state to be considered
+    static constexpr uint32_t MS_LONG = 500;          // duration held to be considered a long press (repeats)
+    static constexpr uint32_t MS_MULTI_TIMEOUT = 500; // duration without a press before the short count is reset
 
-    static uint32_t buttonLastPressed;
-    static uint32_t buttonLastPressedLong;
-    static uint32_t buttonLastPressedShort;
+    static constexpr unsigned STATE_IDLE = 0b111;
+    static constexpr unsigned STATE_FALL = 0b100;
+    static constexpr unsigned STATE_RISE = 0b011;
+    static constexpr unsigned STATE_HELD = 0b000;
 
-    static bool buttonPrevState;  //active high, therefore true as default.
-    static bool buttonIsDown;     //is the button currently being held down?
-    static bool buttonIsDownLong; //is the button currently being held down?
+    // State
+    uint32_t _lastCheck;  // millis of last pin read
+    uint32_t _lastFallingEdge; // millis of last debounced falling edge
+    uint8_t _state; // pin history
+    bool _isLongPress; // true if last press was a long
+    uint8_t _pressCount; // number of short presses before timeout
+public:
+    // Callbacks
+    std::function<void ()>OnShortPress;
```

Comment:
How about if `_pressCount` is argument of OnShortPress callback?

Reply:
I had it this way originally, but passing the count to the handler adds 20 bytes of code to all platforms whether they use the count or not. The BetaFPV 2400 TX target uses the same code space if the value is passed or not so it made sense to not pass it despite being a less desirable pattern.